pkg/hostagent/port.go

```diff
@@ -40,18 +40,24 @@ func hostAddress(rule limayaml.PortForward, guest api.IPPort) string {
 	return host.String()
 }
 
-func (pf *portForwarder) forwardingAddresses(guest api.IPPort) (string, string) {
+func (pf *portForwarder) forwardingAddresses(guest api.IPPort) (hostAddr string, guestAddr string) {
 	for _, rule := range pf.rules {
 		if rule.GuestSocket != "" {
 			continue
 		}
 		if guest.Port < rule.GuestPortRange[0] || guest.Port > rule.GuestPortRange[1] {
 			continue
 		}
+
 		switch {
-		case guest.IP.IsUnspecified():
+		// Move along in case IP versions don't match.
+		// To4() returns `nil` in case it is not an ipv4
+		case (guest.IP.To4() == nil && rule.GuestIP.To4() != nil) ||
+			(guest.IP.To4() != nil && rule.GuestIP.To4() == nil):
```

Comment:
This is an "xor" which can be written as `!=`:

```suggestion
		case (guest.IP.To4() == nil) != (rule.GuestIP.To4() == nil):
```